src/ol_infrastructure/components/services/vault.py

```diff
@@ -134,3 +137,81 @@ def __init__(
                 default_ttl=db_config.default_ttl
             )
         self.register_outputs({})
+
+class OLVaultAWSSecretsEngineConfig(BaseModel):
+    app_name: Text
+    access_key: Text
+    default_lease_ttl_seconds: int = SIX_MONTHS
+    max_lease_ttl_seconds: int = SIX_MONTHS
+    description: Text
+    secret_key: Text
+    path: Text
+    policy_documents: Dict
+    credential_type: str = 'iam_user'
+
+
+class OLVaultAWSSecretsEngine(ComponentResource):
+
+    def __init__(
+            self,
+            engine_config: OLVaultAWSSecretsEngineConfig,
+            opts: ResourceOptions = None
+    ):
+        super().__init__('ol:services:VaultAWSSecretsEngine', engine_config.app_name, None, opts)
+
+        self.aws_secrets_engine = aws.SecretBackend(
+            # TODO verify app_name exists based on Apps class in ol_types
+            f'aws-{engine_config.app_name}',
+            access_key=engine_config.access_key,
+            secret_key=engine_config.secret_key,
+            path=f'aws-{engine_config.app_name}'
+        )
+
+        for role_name, policy in engine_config.policy_documents.items():
+            aws.SecretBackendRole(
+                role_name,
+                backend=f'aws-{engine_config.app_name}',
```

Comment:
Can we use the name or path attribute of the secret backend created on line 162? That creates an explicit dependency chain between the resources.